### PR TITLE
Add S2 keys support to zwave_js config

### LINKED ISSUE #2182: Z-Wave JS addon uses legacy configuration option "networkKey"
## The problem

The Z-Wave JS addon uses the legacy configuration option `networkKey`. According to their documentation this should not be used and instead the new options should be used.
Those are described here: https://zwave-js.github.io/node-zwave-js/#/api/driver?id=zwaveoptions (See the `securityKeys` object)

This causes the following log entries to be printed upon add-on start. As I don't have any S2 accessories at the moment, I cannot verify they are indeed broken with the legacy option.

```
The `networkKey` option is deprecated in favor of `securityKeys` option. To eliminate this warning, move your networkKey into the securityKeys.S0_Legacy option. Refer to the Z-Wave JS docs for more information
[...]
2021-09-13T14:22:56.745Z DRIVER   No network key for S2 configured, communication with secure (S2) devices won't
                                   work!
```

## Environment

- Add-on with the issue: zwave_js
- Add-on release with the issue: latest
- Last working add-on release (if known): N/A 
- Operating environment (OS/Supervised): OS

### PR DESCRIPTION
**Pull request by raman325**
In order to support the new key structure for zwave-js, we need to add three new keys as inputs. We will also migrate keys out of `network_key` into `s0_legacy_key` to better align with this new structure.

This can't be merged until https://github.com/zwave-js/zwave-js-server/pull/353 is released but opening it as a draft for input

fixes #2182

### CODE REVIEW

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:13]: Flush the cache :)

```suggestion
    bashio::addon.option network_key
    bashio::cache.flush_all
```

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:17]: I guess this was done to bypass the caching issue above

```suggestion
    network_key=$(bashio::config "${key}")
```

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:43]: cache bypass?

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (raman325) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:17]: yes, I was not aware of the cache, thanks! I will test this locally before pushing up the changes

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (raman325) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:43]: yup

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:17]: Actually that will not work, let me get back to you later :)

The config is read from disk, not from remote.

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:13]: Not a valid suggestion, sorry about that. The config is read from disk, which doesn't get an update.

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:20]: There is only 1 case, so no need to have an extra construct for that in this case

```suggestion
```

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:23]: There is only 1 case, so no need to have an extra construct for that in this case

```suggestion
    bashio::log.info "Flushing config to disk due to migration"
    bashio::addon.options > "/data/options.json"
```

File: zwave_js/DOCS.md
Comment (MartinHjelmare) [inline zwave_js/DOCS.md:100]: We should clarify that it's only needed to support S2 security. These devices will also work with S0 legacy.

**Comment by raman325** [comment]
is there anything holding this PR back from being merged?

**Comment by MartinHjelmare** [comment]
This looks ok, but I'd like someone that is familiar with bashio to approve before merge.

**Comment by d1b** [comment]
It would be great to see this merged & for S2 support to land.

File: zwave_js/DOCS.md
Comment (kpine) [inline zwave_js/DOCS.md:89]: Maybe "communicate" instead of "reconnect"? The driver can still talk to the device, it's just that the secure commands will fail.

File: zwave_js/DOCS.md
Comment (kpine) [inline zwave_js/DOCS.md:141]:
```suggestion
In previous versions of the addon, this was the only key that was needed.
```

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (kpine) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:39]: Related, it's also highly recommended that all keys be unique, as sharing keys is a security risk. An unknowing user might just copy and paste their S0 key into the new fields, not realizing the consequences. I'm not suggesting that this code should verify all keys are unique (or maybe you want to), as the driver does warn about this (whether it will be seen by a user is another question). Would it be worthwhile to document this recommendation in the security keys section? The driver docs have such a warning:

> Sharing keys between multiple security classes is a security risk!

File: zwave_js/DOCS.md
Comment (raman325) [reply zwave_js/DOCS.md:69]: This is a great point and I have rewritten it a bit, let me know what you think.

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (raman325) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:39]: good idea

File: zwave_js/DOCS.md
Comment (kpine) [reply zwave_js/DOCS.md]: Looks great! 🌟

File: zwave_js/rootfs/usr/share/tempio/zwave_config.conf
Comment (MartinHjelmare) [inline zwave_js/rootfs/usr/share/tempio/zwave_config.conf:15]:
```suggestion
        "S2_Authenticated": "{{ .s2_authenticated_key }}",
        "S2_Unauthenticated": "{{ .s2_unauthenticated_key }}"
```

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

**Review by ludeeus — COMMENTED** [review]
"How to use" and "Configuration" sections needs adjustments

**Comment by MartinHjelmare** [comment]
We need to support the new keys in the integration add-on management too. I'll work on that tonight.

**Comment by MartinHjelmare** [comment]
Here's the core PR to manage S2 keys also in the integration.
https://github.com/home-assistant/core/pull/56783

It can be merged after we merge here and bump the client library used in the core (https://github.com/home-assistant/addons/pull/2188).

File: zwave_js/DOCS.md
Comment (frenck) [inline zwave_js/DOCS.md:46]: We explicitly check against the documentation keys, however, the code only checks against: `2232666D100F795E5BB17F0A1BB7A146`.

**Review by MartinHjelmare — APPROVED** [review]
The docs looks good. Someone with bashio skills should review that part.

File: zwave_js/DOCS.md
Comment (MartinHjelmare) [reply zwave_js/DOCS.md:46]: Should we update the checks?

Comment (MartinHjelmare) [reply zwave_js/DOCS.md:46]: Ok, I'll have a look later tonight, if none beats me to that.

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (MartinHjelmare) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:42]: It's not very nice, but I'm no bash expert. If there's a better way I'll be happy to change it.

**Comment by MartinHjelmare** [comment]
Hmm, I'm getting this error when testing the add-on locally.
```
Error: Invalid networkKey defined
```

File: zwave_js/rootfs/etc/cont-init.d/config.sh
Comment (frenck) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:42]: Let's do this for now and improve later 👍

**Review by frenck — APPROVED** [review]

**Comment by MartinHjelmare** [comment]
I fixed the error. It was just a typo in the config template.

**Review by frenck — APPROVED** [review]

**Pull request merged**
